### ISSUE TITLE
Remember source string for empty/null nodes

### ISSUE THREAD

**Issue by kasp1** [eemeli/yaml#170]
Hi there, I can't seem to be able to stringify `null` as expected, any idea what can be wrong?

repro:

```
const YAML = require('yaml')

let doc = YAML.parse('hi:')

console.log(doc) // 'Object {hi: null}'

// expected string: 'hi:'
console.log(YAML.stringify(doc)) // '? hi'

YAML.scalarOptions.null.nullStr = ''

console.log(YAML.stringify(doc)) // '? hi'

YAML.scalarOptions.null.nullStr = '~'

console.log(YAML.stringify(doc)) // '? hi'
```

**Comment by eemeli** [comment]
The [option](https://eemeli.org/yaml/#options) to do what you want is `{ simpleKeys: true }`. This isn't the first time this is asked, so I think I need to find a way to improve the docs on this, to make it clearer/more discoverable.

**Comment by eemeli** [comment]
Also, this is a bit besides what you're reporting, but the string representation of the `null` value isn't being retained at the AST level like it really ought to. Should fix that too.

**Comment by kasp1** [comment]
Hi, thanks for your advice and for your library. :-)

**Comment by eemeli** [comment]
Fixed in #193.
